### PR TITLE
Download Raspberry Pi Base Firmware with Git

### PR DESCRIPTION
**Pull request by robin-rpr**
Hey there,

This Pull Request aims to clone the Raspberry Pi Base Firmware with `git` instead of `wget`.
It also adds a new Environment Variable called `RPI_FIRMWARE_BRANCH` that references the HEAD of the Firmware Repository to clone.

I subject `stable` as the default value, due to recent (not confirmed) issues with `master`, mentioned in [Issue #5](https://gitlab.com/bboehmke/raspi-alpine-builder/-/issues/5)

Alternative values for `RPI_FIRMWARE_BRANCH` are `master`, `stable` or `next`. See here:
[https://github.com/raspberrypi/firmware/branches/all](https://github.com/raspberrypi/firmware/branches/all)

Cheers!

### CODE REVIEW

File: resources/build.sh [hunk 1]
```diff
@@ -296,27 +298,23 @@ echo ">> Configure boot FS"
...
+    https://github.com/raspberrypi/firmware ${BOOTFS_PATH}/.git
+
+(cd ${BOOTFS_PATH} \
+    && git fetch origin ${RPI_FIRMWARE_BRANCH} \
+    && git branch ${RPI_FIRMWARE_BRANCH} FETCH_HEAD \
+    && git --work-tree=${BOOTFS_PATH} checkout ${RPI_FIRMWARE_BRANCH} \
+    -- `git ls-tree refs/heads/${RPI_FIRMWARE_BRANCH} -r --name-only | grep -E "^boot\/([^\/]+)(\.dat|\.elf)$"`)
+
+mv ${BOOTFS_PATH}/boot/* ${BOOTFS_PATH}
+rm -rf ${BOOTFS_PATH}/boot ${BOOTFS_PATH}/.git
```
Comment (bboehmke) [inline resources/build.sh:313]: Cool idea to git this way.

(I tried a complete shallow clone in the past but it took far to long)

File: resources/build.sh [hunk 2]
```diff
@@ -296,27 +298,23 @@ echo ">> Configure boot FS"
...
 
 # copy linux device trees and overlays to boot
-cp ${ROOTFS_PATH}/boot/dtbs-rpi/*.dtb ${BOOTFS_PATH}/
-cp -r ${ROOTFS_PATH}/boot/dtbs-rpi/overlays ${BOOTFS_PATH}/
+cp ${ROOTFS_PATH}/boot/*.dtb ${BOOTFS_PATH}/
+cp -r ${ROOTFS_PATH}/boot/overlays ${BOOTFS_PATH}/
```
Comment (bboehmke) [inline resources/build.sh]: This is not working for me:
```
cp: can't stat '/work/root_fs/boot/*.dtb': No such file or directory
```

The directory looks like this:
image

**Review by bboehmke — COMMENTED** [review]
with the exception of the dtb files the change looks good

(I will again squash the history on merge)

File: resources/build.sh (hunk 2 above)
Comment (robin-rpr) [reply resources/build.sh]: Oh, thanks! Somehow this worked for me in the actions. Might have been a missing `-e` flag but will investigate this.

Comment (robin-rpr) [reply resources/build.sh]: Here is issue that brought up the change to the paths of the *.dtb files https://github.com/robin-rpr/raspi-alpine-builder/runs/1744298905?check_suite_focus=true#step:6:208

Comment (bboehmke) [reply resources/build.sh]: Is this maybe different based on the version?
I tested with v3.11 and it seams you are using edge.

That would be bad if there is such a change from version to version.

Comment (robin-rpr) [reply resources/build.sh]: Confirming this issue. Seems like the old folder structure will be deprecated by current `edge`. I would suggest modifying the `cp` to add backwards compatibility.

https://github.com/robin-rpr/raspi-alpine-builder/actions/runs/502200372

File: resources/build.sh (hunk 1 above)
Comment (robin-rpr) [reply resources/build.sh:313]: Totally agree! I think there's definitely a way to wrap this up much more slimmer, however I was surprised by the speed myself first.

File: resources/build.sh (hunk 2 above)
Comment (bboehmke) [reply resources/build.sh]: a way to handle both variations would be perfect

Comment (robin-rpr) [reply resources/build.sh]: Give me until the end of next week to fix this up. We are currently re-prioritizing due to some open issues with Rpi builds and plan to support x86_64 first. However we try to keep Arm builds in our backlog and continue working on them later on =)

**Comment by bboehmke** [comment]
This was now implemented in 3419e9e50e5f1bc330c7f27c98e0188d397add39.

**Pull request closed without merging**